### PR TITLE
♻️ Migrate/hotel

### PR DESCRIPTION
**Pull request by lluerich**
Migrated the hotel example backend from [glitch.com](https://glitch.com/~amp-by-example-live) to amp.dev.
It updates the example [here](https://amp.dev/documentation/examples/e-commerce/hotel/?format=websites).

I added some dates in the ```rooms.json``` so that the backend will check if those are available on selected dates.

Part of the migration/improvements in #2054.

### CODE REVIEW

File: examples/source/e-commerce/Hotel/api.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+const express = require('express');
+const {setMaxAge} = require('@lib/utils/cacheHelpers');
+const utils = require('@lib/utils');
+const rooms = require(utils.project.absolute('/examples/static/samples/json/rooms.json'));
+
+// eslint-disable-next-line new-cap
+const examples = express.Router();
+
+examples.all('/rooms', (request, response) => {
```
Comment (sebastianbenz) [inline examples/source/e-commerce/Hotel/api.js]: s/all/get

File: examples/source/e-commerce/Hotel/api.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  setMaxAge(response, 0);
```
Comment (sebastianbenz) [inline examples/source/e-commerce/Hotel/api.js]: max-age can be long as results will always be the same for the given query parameters

Comment (lluerich) [reply examples/source/e-commerce/Hotel/api.js]: Do you have any particular duration in mind? A year?

Comment (sebastianbenz) [reply examples/source/e-commerce/Hotel/api.js]: make it 1d in case we want to change the sample

Comment (lluerich) [reply examples/source/e-commerce/Hotel/api.js]: Alright, PTAL

File: examples/source/e-commerce/Hotel/api.js (hunk 1 above)
Comment (lluerich) [reply examples/source/e-commerce/Hotel/api.js]: PTAL

**Comment by sebastianbenz** [comment]
Thanks!

**Pull request merged**